### PR TITLE
Fix: System Extension for SpringPluginManager

### PR DESCRIPTION
**Pull request by lhanda**
fixes for the followings:
1) system extension not the same instance as the one registered in the application context of the SpringPluginManager
2) plugin manager registering system extension as beans again even if beans already existed causing more than one instance of the same class bean to exist

Scenario as follows:
- defined a SpringPluginManager
- defined a system extension but also annotated it using Spring Framework annotations, eg. @Service, @Component, etc.
- Spring Framework will create the system extension but inside SpringExtensionFactory, it will create an instance instead of retrieving it from the Spring application context
- ExtensionInjector will also re-register already registered extensions as beans in the Spring application context

### CODE REVIEW

File: pf4j-spring/src/main/java/org/pf4j/spring/ExtensionsInjector.java [hunk 1]
```diff
@@ -31,17 +31,17 @@
 
     private static final Logger log = LoggerFactory.getLogger(ExtensionsInjector.class);
 
-    protected final PluginManager pluginManager;
+    protected final SpringPluginManager springPluginManager;
```
Comment (decebals) [inline pf4j-spring/src/main/java/org/pf4j/spring/ExtensionsInjector.java:34]: Is this change really needed?

**Comment by decebals** [comment]
@lhanda 

Can you solve the conflict (merge/rebase against master)?

**Comment by decebals** [comment]
To be honest I don't use Spring (`pf4j-spring`) in my projects.
From PR's description the fixes are clear but I don't see reactions from community so, I don't know if the PR is valuable enough for community. 
In the end, I think that I will try to merge it.

File: pf4j-spring/src/main/java/org/pf4j/spring/ExtensionsInjector.java (hunk 1 above)
Comment (lhanda) [reply pf4j-spring/src/main/java/org/pf4j/spring/ExtensionsInjector.java:34]: From my assessment of the ExtensionsInjector class, it seems that the class is primarily used for registration of extensions to the application context of the SpringPluginManager only. Instead of passing in the SpringPluginManager as a boxed up parameter, I thought we should pass in as SpringPluginManager instead so that we could retrieve the application context, to make use of the 'getBeansOfType' method to retrieve all registered instances of the extension class. Another alternative is to use the 'getBean' method of the beanFactory parameter but that involves catching 'NoSuchBeanDefinitionException' and registering the extension in the handler.

**Comment by lhanda** [comment]
> @lhanda
> 
> Can you solve the conflict (merge/rebase against master)?

Done

**Comment by decebals** [comment]
Thanks for your contribution!
